Pulling images doesn't seem to work

dobi.yaml:

```
...

image=go-swagger:
    image: "quay.io/goswagger/swagger:0.6.0"
    pull: once

job=go-swagger-validate:
    use: go-swagger
    command: "validate /code/swagger.yaml"
    ...
```

```
$ dobi go-swagger-validate
[image:build go-swagger] quay.io/goswagger/swagger:0.6.0 Building
[ERROR] Failed to execute task "go-swagger:build": missing remote repository e.g. 'github.com/user/repo'
```

There's no example of how to do this, so I'm guessing this is correct. Either I've got it wrong and an example is needed, or it's actually broken.

Thanks. Upgraded to 0.6, but it seems to require auth for an image which doesn't (I think?):

```
ben@computer ~/p/docker-api-reference [master]
± % dobi go-swagger-validate                                                        !97
[WARN] [image:pull go-swagger] quay.io/goswagger/swagger Failed to get image record: stat /Users/ben/p/docker-api-reference/.dobi/images/quay.io goswagger swagger:0.6.0: no such file or directory
[ERROR] Failed to execute task "go-swagger:pull": API error (401): Get https://quay.io/v2/goswagger/swagger/manifests/0.6.0: unauthorized: authentication required


ben@computer ~/p/docker-api-reference [master]
± % docker pull quay.io/goswagger/swagger:0.6.0                                     !98
0.6.0: Pulling from goswagger/swagger
e110a4a17941: Already exists
3f334f20cdfb: Already exists
6effbd395226: Already exists
Digest: sha256:3d1f8e34a70a528a421a921a0c3bc9ada5bb16aad4404d70dd40d8bee1702573
Status: Image is up to date for quay.io/goswagger/swagger:0.6.0
```

--- a/config/image.go
+++ b/config/image.go
@@ -7,6 +7,7 @@ import (
 	"time"
 
 	"github.com/dnephin/dobi/execenv"
+	docker "github.com/fsouza/go-dockerclient"
 )
 
 // ImageConfig An **image** resource provides actions for working with a Docker
@@ -26,8 +27,9 @@ import (
 //           url: http://example.com/foo
 //
 type ImageConfig struct {
-	// Image The name of the **image** without any tags
-	Image string `config:"required"`
+	// Image The name of the **image** without a tag. Tags must be specified
+	// in the **tags** field.
+	Image string `config:"required,validate"`
 	// Dockerfile The path to the ``Dockerfile`` used to build the image. This
 	// path is relative to the **context**.
 	Dockerfile string
@@ -68,8 +70,6 @@ func (c *ImageConfig) Dependencies() []string {
 
 // Validate checks that all fields have acceptable values
 func (c *ImageConfig) Validate(path Path, config *Config) *PathError {
-	// TODO: validate no tag on image name
-
 	if err := c.validateBuildOrPull(); err != nil {
 		return PathErrorf(path, err.Error())
 	}
@@ -85,6 +85,16 @@ func (c *ImageConfig) validateBuildOrPull() error {
 		c.Dockerfile = "Dockerfile"
 	case c.Context == "" && c.Dockerfile != "":
 		c.Context = "."
+	}
+	return nil
+}
+
+// ValidateImage validates the image field does not include a tag
+func (c *ImageConfig) ValidateImage() error {
+	_, tag := docker.ParseRepositoryTag(c.Image)
+	if tag != "" {
+		return fmt.Errorf(
+			"Tag %q must be specified in the `tags` field, not in `image`", tag)
 	}
 	return nil
 }

--- a/tasks/image/inspect.go
+++ b/tasks/image/inspect.go
@@ -2,8 +2,11 @@ package image
 
 import (
 	"fmt"
+	"net/url"
+	"strings"
 
 	"github.com/dnephin/dobi/config"
+	"github.com/dnephin/dobi/logging"
 	"github.com/dnephin/dobi/tasks/context"
 	docker "github.com/fsouza/go-dockerclient"
 )
@@ -31,6 +34,23 @@ func GetCanonicalTag(ctx *context.ExecuteContext, conf *config.ImageConfig) stri
 }
 
 func parseRepo(image string) (string, error) {
-	// TODO: how is this supposed to work?
-	return defaultRepo, nil
+	repo, _ := docker.ParseRepositoryTag(image)
+	if !strings.HasPrefix(repo, "http") {
+		repo = "https://" + repo
+	}
+
+	addr, err := url.Parse(repo)
+	if err != nil {
+		return "", fmt.Errorf("Failed to parse repo name from %q: %s", repo, err)
+	}
+
+	if addr.Host == "" {
+		logging.Log.Debugf("Using default registry %q", defaultRepo)
+		return defaultRepo, nil
+	}
+
+	// TODO: what about v2?
+	addr.Path = "v1"
+	logging.Log.Debugf("Using registry %q", addr)
+	return addr.String(), nil
 }

